library/core/src/iter/adapters/mod.rs

```diff
@@ -1687,6 +1687,110 @@ impl<I: Iterator> Peekable<I> {
     {
         self.next_if(|next| next == expected)
     }
+
+    /// Creates an iterator that consumes elements until predicate is
+    /// true, without consuming the last matching element.
+    ///
+    /// `until()` takes a closure as an argument. It will call this
+    /// closure on each element of the iterator, and consume elements
+    /// until it returns true.
+    ///
+    /// After true is returned, until()'s job is over, and the iterator
+    /// is fused.
+    ///
+    /// This is the exact opposite of [`skip_while`].
+    ///
+    /// # Example
+    /// Consume numbers until you find a '5'.
+    /// ```
+    /// #![feature(peekable_next_if)]
+    /// let mut iter = (0..10).peekable();
+    /// assert_eq!(iter.until(|&x| x == 5).collect::<String>(), "1234".to_string());
+    /// assert_eq!(iter.next(), Some(5));
+    /// ```
+    /// [`skip_while`]: trait.Iterator.html#method.skip_while
+    #[unstable(feature = "peekable_next_if", issue = "72480")]
+    pub fn until<P: FnMut(&I::Item) -> bool>(&mut self, func: P) -> Until<'_, I, P> {
+        Until::new(self, func)
+    }
+}
+
+/// An iterator that iterates over elements until `predicate` returns `false`.
+///
+/// This `struct` is created by the [`until`] method on [`Peekable`]. See its
+/// documentation for more.
+///
+/// [`until`]: trait.Peekable.html#until
+/// [`Iterator`]: trait.Iterator.html
```

Comment:
cross doc link, we probably don't need this. @jyn514 would be happy if you remove this